### PR TITLE
Fix handling of partially flattened config keys 

### PR DESCRIPTION
**Pull request by strawgate**
https://github.com/elastic/crawler/issues/272

Recursively unnest flattened keys to ensure partially flat keys work:
```
key:
  sub.key: 5
```

#### Pre-Review Checklist
- [x] This PR does NOT contain credentials of any kind, such as API keys or username/passwords (double check `crawler.yml.example` and `elasticsearch.yml.example`)
- [x] This PR has a meaningful title
- [x] This PR links to all relevant GitHub issues that it fixes or partially addresses
    - If there is no GitHub issue, please create it. Each PR should have a link to an issue
- [x] this PR has a thorough description
- [x] Covered the changes with automated tests
- [x] Tested the changes locally
- [ ] Added a label for each target release version (example: `v0.1.0`)
- [x] Considered corresponding documentation changes
- [x] Contributed any configuration settings changes to the configuration reference
- [ ] Ran `make notice` if any dependencies have been added

#### Changes Requiring Extra Attention

- [ ] Security-related changes (encryption, TLS, SSRF, etc)
- [ ] New external service dependencies added.

### CODE REVIEW

**Comment by mattnowzari** [comment]
buildkite test this

File: spec/lib/crawler/cli/helpers_spec.rb [hunk 1]
```diff
@@ -0,0 +1,64 @@
+# Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+# or more contributor license agreements. Licensed under the Elastic License 2.0;
+# you may not use this file except in compliance with the Elastic License 2.0.
+#
+
+# frozen_string_literal: true
+
```
Comment (mattnowzari) [inline spec/lib/crawler/cli/helpers_spec.rb:7]: Not a nit, but a kind request if you have the free cycles to dedicate to this!

We already have a spec file to test the `Helpers` class under `spec/lib/crawler/api/helper_spec.rb`. To be honest, the pre-existing spec file is in the wrong place to begin with (you've done the correct thing by putting yours under `crawler/cli/`. Do you think you could move some of those tests from the pre-existing spec file into this one?

When you do, please make sure to use the YAML fixtures that already exist under `/spec/fixtures`. Feel free to edit or even create new fixtures that include this partially flattened edge case.

I definitely was not an RSpec master when I wrote those tests, so use your discretion and best judgement to improve/refactor any of those tests as you see best.

Thank you so much🙏

**Review by mattnowzari — CHANGES_REQUESTED** [review]
Thank you for catching this edge case and getting a fix together! 😄

I just have a request regarding the test cases.

File: spec/lib/crawler/cli/helpers_spec.rb (hunk 1 above)
Comment (mattnowzari) [reply spec/lib/crawler/cli/helpers_spec.rb:7]: Given that this PR is a bugfix, I would like to prioritize this over the secrets PR if that is OK. We can stick with moving the pre-existing tests inside of this PR, as opposed to creating a new PR, etc - this way other potential reviewers can keep context of the bug fix work, the related tests, etc in one place.

FYI, while this is definitely a bug, it's an edge case and not _entirely_ breaking functionality, so please don't feel rushed to get this work out 🙂 If it becomes an issue for users, the workaround would be to not use nested flat YAML until we get this fixed!

Comment (strawgate) [reply spec/lib/crawler/cli/helpers_spec.rb:7]: sounds good!

Comment (strawgate) [reply spec/lib/crawler/cli/helpers_spec.rb:7]: @mattnowzari I have combined the tests as requested and added a new fixture in a commit

I then did a slight refactor to improve clarity in helpers and a refactor of the test to better utilize rspec features. I did this in its own commit

If you don't like the refractor let me know and I'll undo the commit most recent

**Comment by mattnowzari** [comment]
buildkite test this

**Comment by mattnowzari** [comment]
buildkite test this

**Review by mattnowzari — APPROVED** [review]
LGTM! Thank you for the fix and for merging the test cases into one file 🚀
